railties/lib/rails/generators/rails/app/templates/Dockerfile.tt

```diff
@@ -21,10 +21,10 @@ RUN apt-get update -qq && \
 
 <% if using_node? -%>
 # Install JavaScript dependencies
-ARG NODE_VERSION=<%= dockerfile_node_version %>
+ARG NODE_VERSION=<%= node_version %>
 ARG YARN_VERSION=<%= dockerfile_yarn_version %>
 
-ENV PATH=$PATH:/usr/local/node/bin
+ENV PATH=/usr/local/node/bin:$PATH
```

Comment:
I missed that the system node might be pulled via `apt-get install node-gyp`, so we need to give precedence to the custom node.

That should fix https://github.com/rails/rails/pull/47460#issuecomment-1440342337 cc @rubys